java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/analyser/deploy/git/Git.java

```diff
@@ -0,0 +1,102 @@
+package com.redhat.hacbs.container.analyser.deploy.git;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.lib.StoredConfig;
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.PushResult;
+import org.eclipse.jgit.transport.RemoteRefUpdate;
+
+import io.quarkus.logging.Log;
+
+public abstract class Git {
+
+    protected CredentialsProvider credentialsProvider;
+
+    public abstract void create(String name)
+            throws IOException, URISyntaxException;
+
+    public abstract void add(Path path, String commit)
+            throws IOException;
+
+    protected void pushRepository(Path path, String httpTransportUrl, String commit) {
+        try (var jGit = org.eclipse.jgit.api.Git.init().setDirectory(path.toFile()).call()) {
+            // TODO: Discuss tag naming - should a rebuild produce the same results ? In which case why use a different tag?
+            //   As JBS might have changed the code? We could append the image-id i.e.
+            //   apache/commons-net.git@rel/commons-net-3.9.0    DependencyBuildStateComplete   75ecd81c7a2b384151c990975eb1dd10
+            //   Tag would be
+            //      rel/commons-net-3.9.0-75ecd81c7a2b384151c990975eb1dd10
+            var tagName = jGit.describe().setTags(true).setTarget(commit).call();
```

Comment:
I think potentially we want two different additional tags, and two different ID concepts to go with it.

I think we want two different IDs:
- A stable build id (e.g. the DependencyBuild name)
- A completely unique UID for the build

We could then create two tags with both IDs appended or prepended. The idea would be that if we ever rebuild something the stable ID would be overwritten, but the unique one would always exist. We could do the same for image deployments, so that even when there are rebuilds we never loose anything.